azure/sdk-identity-resources-storage/Program.cs

```diff
@@ -0,0 +1,137 @@
+            Console.WriteLine($"Done creating account {storageAccountName}.");
+
+            return storageAccountName;
+        }
+
+        private static async Task<string> GetStorageAccountName(StorageManagementClient storageManagementClient)
+        {
+            bool foundAvailableName = false;
+            string storageAccountName = string.Empty;
+
+            while (!foundAvailableName)
+            {
+                storageAccountName = RandomName("storage", 20);
+                Response<CheckNameAvailabilityResult> availability =
+                    await storageManagementClient.StorageAccounts.CheckNameAvailabilityAsync(
+                        new StorageAccountCheckNameAvailabilityParameters(storageAccountName));
+
+                if (availability.Value.NameAvailable.GetValueOrDefault())
+                {
+                    foundAvailableName = true;
+                }
+            }
+            return storageAccountName;
+        }
+
+        private static async Task UploadBlobToStorageAccountUsingClientConnectionString(StorageManagementClient storageManagementClient, string resourceGroupName, string storageName)
+        {
+            Console.WriteLine("Creating a container and uploading a blob...");
+
+            string uploadFileName = "dotnet-bot.png";
+            string containerName = "images";
+            string connectionString = await GetStorageConnectionString(storageManagementClient, resourceGroupName, storageName);
+
+            var containerClient = new BlobContainerClient(connectionString, containerName);
+            await containerClient.CreateIfNotExistsAsync(publicAccessType: PublicAccessType.Blob);
+
+            BlobClient blobClient = containerClient.GetBlobClient(uploadFileName);
+            await blobClient.UploadAsync(uploadFileName);
+
+            Console.WriteLine($"Your blob is at https://{storageName}.blob.core.windows.net/{containerName}/{uploadFileName}");
+        }
+
+
+        private static async Task<string> GetStorageConnectionString(StorageManagementClient storageManagementClient, string resourceGroupName, string storageAccountName)
+        {
+            Response<StorageAccountListKeysResult> keys = await storageManagementClient.StorageAccounts.ListKeysAsync(resourceGroupName, storageAccountName);
+            StorageAccountKey storageKey = keys.Value.Keys[0];
+            return $"DefaultEndpointsProtocol=https;AccountName={storageAccountName};AccountKey={storageKey.Value};EndpointSuffix=core.windows.net;";
+        }
+
+        private static async Task<string> CreateResourceGroup(ResourcesManagementClient resourcesManagementClient)
+        {
+            string resourceGroupName = RandomName("rg", 20);
+            Console.WriteLine($"Creating resource group {resourceGroupName}...");
+            await resourcesManagementClient.ResourceGroups.CreateOrUpdateAsync(resourceGroupName, new ResourceGroup("West US"));
```

Comment:
Nit - use a constant for `West US` or pass it in.

Reply:
I agree, since this the region `"West US"` is used in multiple places, perhaps make it a class scoped const?

```csharp
class Program
{
    const string ResourceRegion = "West US";

    ...
}
```